google/resource_dns_record_set.go

```diff
@@ -55,6 +57,101 @@ func resourceDnsRecordSet() *schema.Resource {
 	}
 }
 
+// This catches most invalid DNS names, though it might not catch some invlaid non-ASCII names
+// it does not implement the TOASCII functionality defined in https://tools.ietf.org/html/rfc3454
+// See:
+// https://tools.ietf.org/html/rfc1034 DOMAIN NAMES - CONCEPTS AND FACILITIES
+// https://tools.ietf.org/html/rfc3490 Internationalizing Domain Names in Applications (IDNA)
+// https://tools.ietf.org/html/rfc3454 Preparation of Internationalized Strings ('stringprep')
+func validate_dns_rrset_name(name string) error {
+	if len(name) > 255 {
+		return fmt.Errorf("DNS name length %d greater than 254", len(name))
+	}
+
+	re, err := regexp.Compile("[\u002e\u3002\uFF0E\uFF61]$")
```

Comment:
Can you add comments as to what each of these unicode codes is and why they need to be specified this way rather than just doing something like:
```
if !strings.HasSuffix(name, ".") {
  return fmt.Errorf(...)
}
```

Reply:
Added. Specifically, these are required for internationalized domain names. 

Whenever dots are used as label separators, the following characters MUST be recognized as dots: U+002E (full stop), U+3002 (ideographic full stop), U+FF0E (fullwidth full stop), U+FF61 (halfwidth ideographic full stop).

Reply:
Done.